src/Service/VendorService/AbstractBaseVendorService.php

```diff
@@ -164,60 +166,94 @@ protected function updateOrInsertMaterials(array &$identifierImageUrlArray, stri
+     *   Sources repository
+     * @param string $identifierType
+     *   The type of identifiers in to be processed
+     *
+     * @return array
+     *   Array containing two arrays with identifiers for updated and inserted sources
+     *
+     * @throws IllegalVendorServiceException
+     * @throws UnknownVendorServiceException
+     * @throws QueryException
+     */
+    private function processBatch(array $batch, SourceRepository $sourceRepo, string $identifierType): array
+    {
+        // Split into to results arrays (updated and inserted).
+        $updatedIdentifiers = [];
+        $insertedIdentifiers = [];
+
+        // Load batch from database to enable updates.
+        $sources = $sourceRepo->findByMatchIdList($identifierType, $batch, $this->getVendor());
+
+        foreach ($batch as $identifier => $imageUrl) {
+            if (array_key_exists($identifier, $sources)) {
+                $source = $sources[$identifier];
+                ++$this->totalUpdated;
+                $updatedIdentifiers[] = $identifier;
+            } else {
+                $source = new Source();
+                $this->em->persist($source);
+                ++$this->totalInserted;
+                $insertedIdentifiers[] = $identifier;
             }
 
-            $this->em->flush();
-            $this->em->clear('App\Entity\Source');
+            $source->setMatchType($identifierType)
+                ->setMatchId($identifier)
+                ->setVendor($this->vendor)
+                ->setDate(new \DateTime())
+                ->setOriginalFile($imageUrl);
 
-            // Send event with the last batch to the job processors.
-            if ($this->queue) {
-                if (!empty($insertedIds)) {
-                    $event = new VendorEvent(VendorState::INSERT, $insertedIds, $identifierType, $this->vendor->getId());
-                    $this->dispatcher->dispatch($event::NAME, $event);
-                }
-                if (!empty($updatedIds)) {
-                    $event = new VendorEvent(VendorState::UPDATE, $updatedIds, $identifierType, $this->vendor->getId());
-                    $this->dispatcher->dispatch($event::NAME, $event);
-                }
-
-                // @TODO: DELETED event???
+            ++$this->totalIsIdentifiers;
+        }
+
+        $this->em->flush();
+        $this->em->clear();
+
+        gc_collect_cycles();
+
+        return [$updatedIdentifiers, $insertedIdentifiers];
+    }
+
+    /**
+     * Send events to the job queue with identifiers to process.
+     *
+     * @param array $updatedIdentifiers
+     *   Updated identifiers
+     * @param array $insertedIdentifiers
+     *   Inserted identifiers
+     * @param string $identifierType
+     *   The type of identifiers in to be processed
+     */
+    private function sendCoverImportEvents(array $updatedIdentifiers, array $insertedIdentifiers, string $identifierType): void
+    {
+        if ($this->queue) {
```

Comment:
Why this test?

Reply:
We have a `queue` option for the vendor command that can be set to false if you don't want the imprt to go to the queue

Reply:
👍